src/Core/Tools/SendFeatures/Commands/Interfaces/INonAnonymousSendCommand.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Bit.Core.Tools.Entities;
+using Bit.Core.Tools.Models.Data;
+
+namespace Bit.Core.Tools.SendFeatures.Commands.Interfaces;
+
+/// <summary>
+/// NonAnonymousSendCommand interface provides methods for managing non-anonymous Sends.
+/// </summary>
+public interface INonAnonymousSendCommand
+{
+    /// <summary>
+    /// Saves a Send to the database.
+    /// </summary>
+    /// <param name="send">Send object that is to be saved to database</param>
+    /// <returns>Task object for async operations</returns>
+    Task SaveSendAsync(Send send);
+
+    /// <summary>
+    /// Save File to the database.
+    /// </summary>
+    /// <param name="send">Send object that is to be saved to database</param>
+    /// <param name="data">SendFileData to be saved to database</param>
+    /// <param name="fileLength">Length of file help with saving to database</param>
+    /// <returns>Task object for async operations with file upload url</returns>
+    Task<string> SaveFileSendAsync(Send send, SendFileData data, long fileLength);
+    /// <summary>
+    /// Upload a file to an existing Send.
+    /// </summary>
+    /// <param name="stream">Stream of file to be uploaded</param>
```

Comment:
🤔 This should probably indicate what happens when a stream is given to the method whose `.Position > 0`.

Reply:
Added this to param summary

Reply:
You noted what it assumes, not how it behaves. This is incorrect because you need to read the code to understand what the effect of that assumption is.

How it behaves is **it saves the file from the current stream position**.